### PR TITLE
#162949217 Articles CRUD functionality

### PR DESCRIPTION
**Pull request by nesh-dev**
#### What does this PR do?

Add Articles CRUD functionality for 

#### Description of Task to be completed?

* An authorized user can create an article.
* A user can get a single article.
* A user can get a list of all articles.
* An author can update their article 
* An author can delete their article

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git pull branch ft-create-articles-162949217
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file in the same level as the venv file and add the following:
      ```
           source venv/bin/activate
           export SECRET_KEY=""
           export DATABASE_URL=""
           export DJANGO_SETTINGS_MODULE='authors.settings'
           export CLOUDINARY_NAME=''
           export CLOUDINARY_KEY=''
           export CLOUDINARY_SECRET=''
           export HOST_PASSWORD=""
           export HOST_USER=""
           export SMTP_HOST=''
           export EMAIL_PORT=''
      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```
7. To test you need to register and login via postman or your browser to and use swagger.
* Register endpoint `/api/users`
* Login endpoint `/api/users/login`

8. Test the following endpoints while authenticated via JWT.
* POST `api/articles`
* GET ALL `api/articles`
* GET single `api/articles/<slug:slug>`
*  PUT `api/articles/<slug:slug>`
* DEL `api/articles/<slug:slug>`
#### Any background context you want to provide?
You need the above keys from the admin.

#### What are the relevant pivotal tracker stories?
[#162949217](https://www.pivotaltracker.com/story/show/162949217)
#### Screenshots (if appropriate)
screenshot 2019-01-24 at 08 41 25
screenshot 2019-01-24 at 08 42 37

#### Questions:

### CODE REVIEW

**Review by MbugwaSami — DISMISSED** [review]
looks good to me

**Review by kwanj-k — COMMENTED** [review]
Good work @nesh-dev 
But your review app is not working
Kindly add the.env settings on your PR to heroku so we can test it

**Comment by nesh-dev** [comment]
> Good work @nesh-dev
> But your review app is not working
> Kindly add the.env settings on your PR to heroku so we can test it

@kwanj-k  I have added the env variables. It should work now. Thanks for the feedback.

**Review by kwanj-k — DISMISSED** [review]

File: authors/tests/test_articles.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            reverse(
+                'articles:detail_article',
+                kwargs={'slug': 'another-post'},
+            ),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertIn(response.data, error_messages['unauthorised'])
+
+    def test_update_article_none_author(self):
```
Comment (Allan690) [inline authors/tests/test_articles.py:192]: Update the docstring here

File: authors/tests/test_articles.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                'articles:detail_article',
+                kwargs={'slug': 'another-post'},
+            ),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertIn(response.data, error_messages['unauthorised'])
+
+    def test_update_article_null_data(self):
+        """Test updating an article """
```
Comment (Allan690) [inline authors/tests/test_articles.py]: Kindly update the docstrings accordingly

File: authors/tests/test_articles.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                'articles:detail_article',
+                kwargs={'slug': 'another-posts'},
+            ),
+            content_type='application/json'
+        )
+        self.assertIn(error_messages['article_404'], response.data)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_unauthorized(self):
+        """Test delete while unauthorized  """
```
Comment (Allan690) [inline authors/tests/test_articles.py]: Make the docstrings more descriptive of the test being run

**Review by Allan690 — DISMISSED** [review]

**Review by kwanj-k — DISMISSED** [review]

**Review by MbugwaSami — APPROVED** [review]

**Review by kelvinwm — APPROVED** [review]

**Review by Allan690 — APPROVED** [review]

**Review by indungu — APPROVED** [review]
Nice work

**Pull request merged**